x/delayedack/keeper/invariants_test.go

```diff
@@ -0,0 +1,284 @@
+		suite.Require().Nil(err)
+		break
+	}
+
+	// check invariant
+	msg, bool := keeper.AllInvariants(suite.App.DelayedAckKeeper)(suite.Ctx)
+	suite.Require().False(bool, msg)
+}
+
+func (suite *DelayedAckTestSuite) TestRollappPacketsCasesInvariant() {
+	suite.SetupTest()
+	ctx := suite.Ctx
+	rollapp := "rollapp1"
+	cases := []struct {
+		name             string
+		rollappId        string
+		stateInfo        *rollapptypes.StateInfo
+		stateInfo2       *rollapptypes.StateInfo
+		packet           commontypes.RollappPacket
+		packet2          commontypes.RollappPacket
+		expectedIsBroken bool
+	}{
+		{
+			"successful invariant check",
+			rollapp,
+			&rollapptypes.StateInfo{
+				StateInfoIndex: rollapptypes.StateInfoIndex{
+					RollappId: rollapp,
+					Index:     1,
+				},
+				StartHeight: 1,
+				NumBlocks:   10,
+				Status:      commontypes.Status_FINALIZED,
+			},
+			&rollapptypes.StateInfo{
+				StateInfoIndex: rollapptypes.StateInfoIndex{
+					RollappId: rollapp,
+					Index:     2,
+				},
+				StartHeight: 10,
+				NumBlocks:   10,
+				Status:      commontypes.Status_PENDING,
+			},
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_FINALIZED,
+				ProofHeight: 5,
+				Packet: &channeltypes.Packet{
+					SourcePort:         "testSourcePort",
+					SourceChannel:      "testSourceChannel",
+					DestinationPort:    "testDestinationPort",
+					DestinationChannel: "testDestinationChannel",
+					Data:               []byte("testData"),
+					Sequence:           uint64(1),
+				},
+			},
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_PENDING,
+				ProofHeight: 15,
+				Packet: &channeltypes.Packet{
+					SourcePort:         "testSourcePort",
+					SourceChannel:      "testSourceChannel",
+					DestinationPort:    "testDestinationPort",
+					DestinationChannel: "testDestinationChannel",
+					Data:               []byte("testData"),
+					Sequence:           uint64(2),
+				},
+			},
+			false,
+		},
+		{
+			"error non-finalized packet",
+			rollapp,
+			&rollapptypes.StateInfo{
+				StateInfoIndex: rollapptypes.StateInfoIndex{
+					RollappId: rollapp,
+					Index:     1,
+				},
+				StartHeight: 1,
+				NumBlocks:   10,
+				Status:      commontypes.Status_FINALIZED,
+			},
+			&rollapptypes.StateInfo{
+				StateInfoIndex: rollapptypes.StateInfoIndex{
+					RollappId: rollapp,
+					Index:     2,
+				},
+				StartHeight: 10,
+				NumBlocks:   10,
+				Status:      commontypes.Status_FINALIZED,
+			},
+			commontypes.RollappPacket{
+				RollappId:   rollapp,
+				Status:      commontypes.Status_FINALIZED,
+				ProofHeight: 5,
+				Packet: &channeltypes.Packet{
+					SourcePort:         "testSourcePort",
+					SourceChannel:      "testSourceChannel",
+					DestinationPort:    "testDestinationPort",
+					DestinationChannel: "testDestinationChannel",
+					Data:               []byte("testData"),
```

Comment:
this is duplicated a lot, can we extract a function to create this (with a variable sequence number)

Reply:
done